src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -46,6 +77,12 @@ public AuthorizeFilter(AuthorizationPolicy policy)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            Policy = Policy ?? await AuthorizationPolicy.CombineAsync(PolicyProvider, AuthorizeData);
```

Comment:
Okay so new behavior will be:
1. If AuthorizePolicy ctor is used, its cached and used always.
2. Otherwise, policy provider Combine is called with IAuthorizeData every time.